fe/fe-core/src/main/java/com/starrocks/connector/hive/CacheUpdateProcessor.java

```diff
@@ -85,17 +121,67 @@ public void refreshTable(String dbName, Table table, boolean onlyCachedPartition
         }
     }
 
-    public void refreshTableWithExecutor(Table table, boolean onlyCachedPartitions, ExecutorService executor) {
+    public void refreshTableBackground(Table table, boolean onlyCachedPartitions, ExecutorService executor) {
         HiveMetaStoreTable hmsTbl = (HiveMetaStoreTable) table;
-        metastore.refreshTable(hmsTbl.getDbName(), hmsTbl.getTableName(), onlyCachedPartitions);
-        refreshRemoteFiles(hmsTbl.getTableLocation(), Operator.UPDATE, getExistPaths(hmsTbl), onlyCachedPartitions,
-                executor);
+        if (metastore.refreshTableBackground(hmsTbl.getDbName(), hmsTbl.getTableName(), onlyCachedPartitions)) {
+            Map<BasePartitionInfo, Partition> updatedPartitions = getUpdatedPartitions(hmsTbl);
+            if (!updatedPartitions.isEmpty()) {
+                // update partition remote files cache
+                List<String> updatedPaths = updatedPartitions.values().stream().map(Partition::getFullPath)
+                        .map(path -> path.endsWith("/") ? path : path + "/")
+                        .collect(Collectors.toList());
+                refreshRemoteFilesBackground(hmsTbl.getTableLocation(), updatedPaths, onlyCachedPartitions, executor);
+
+                LOG.info("{}.{}.{} partitions has updated, updated partition size is {}, " +
+                                "refresh partition and file success", hmsTbl.getCatalogName(), hmsTbl.getDbName(),
+                        hmsTbl.getTableName(), updatedPartitions.size());
+            }
+
+            // update partitionUpdatedTimes
+            updatedPartitions.forEach((key, value) -> partitionUpdatedTimes.put(key, value.getModifiedTime()));
+            partitionUpdatedTimes.keySet().removeIf(basePartitionInfo -> !getCachedTableNames().contains(
+                    HiveTableName.of(basePartitionInfo.dbName, basePartitionInfo.tableName)));
+        }
     }
 
     public Set<HiveTableName> getCachedTableNames() {
         return ((CachingHiveMetastore) metastore).getCachedTableNames();
     }
 
+    private Map<BasePartitionInfo, Partition> getUpdatedPartitions(HiveMetaStoreTable table) {
+        String dbName = table.getDbName();
+        String tblName = table.getTableName();
+
+        Map<BasePartitionInfo, Partition> toCheckUpdatedPartitionInfoMap = Maps.newHashMap();
+        if (table.isUnPartitioned()) {
+            Partition partition = metastore.getPartition(dbName, tblName, Lists.newArrayList());
+            BasePartitionInfo partitionInfo = new BasePartitionInfo(dbName, tblName, tblName);
+            toCheckUpdatedPartitionInfoMap.put(partitionInfo, partition);
+        } else {
+            List<String> partitionNames = metastore.getPartitionKeys(dbName, tblName, false);
+            Map<String, Partition> partitions =  metastore.getPartitionsByNames(dbName, tblName, partitionNames);
```

Comment:
I'm afraid we will get all partition here, why not just get partitions present in cache?